cdap-client/src/main/java/co/cask/cdap/client/NamespaceClient.java

```diff
@@ -61,9 +62,18 @@ public NamespaceClient(ClientConfig config) {
    * @throws UnAuthorizedAccessTokenException if the request is not authorized successfully in the gateway server
    */
   public List<NamespaceMeta> list() throws IOException, UnAuthorizedAccessTokenException {
-    HttpResponse response = restClient.execute(HttpMethod.GET, config.resolveURL("namespaces"),
-                                               config.getAccessToken());
-    return ObjectResponse.fromJsonBody(response, new TypeToken<List<NamespaceMeta>>() { }).getResponseObject();
+    // TODO: remove the following apiVersion set/reset logic all APIs are migrated to v3
```

Comment:
Agreed, let's think about it. I'm just concerned that allowing users to set API versions might lead to issues if for example a user tries to use a v3 version of one API with a v2 version of another API.

Agree though that we should think of improving this.